js/common/lib/tensor-impl.ts

```diff
@@ -233,24 +247,24 @@ export class Tensor implements TensorInterface {
 
     // Updating the pointer assignments based on the output tensor format
     if (outputformat === 'RGBA') {
-      aTensorPointer = offset * 3;
+      aTensorPointer = stride * 3;
     } else if (outputformat === 'RBG') {
       rTensorPointer = 0;
-      bTensorPointer = offset;
-      gTensorPointer = offset * 2;
+      bTensorPointer = stride;
+      gTensorPointer = stride * 2;
     } else if (outputformat === 'BGR') {
       bTensorPointer = 0;
-      gTensorPointer = offset;
-      rTensorPointer = offset * 2;
+      gTensorPointer = stride;
+      rTensorPointer = stride * 2;
     }
 
-    for (let i = 0; i < offset;
+    for (let i = 0; i < stride;
          i++, rImagePointer += step, bImagePointer += step, gImagePointer += step, aImagePointer += step) {
```

Comment:
is it assuming NCHW for the image data here?

Reply:
Yeah, right now the default if NCHW. I fixed the documentation to add a TODO for NHWC.